postgresql versions moved to common facts

vjrj opened the pull request:
Following my tests with #598 and #599 I moved the postgresql/postgis versions to common / set-facts.yml.

Also related with #597.

sbearcsiro commented on ansible/spatial.yml:
```diff
@@ -5,7 +5,7 @@
     - tomcat
     - webserver
-    - {role: postgresql, pg_service: 'postgresql-9.6', pg_version: '9.6', postgis_version: '2.4'}
+    - postgresql
```
instead of removing the version, why not add `spatial_pg_version` and `sptaial_postgis_version` and then override them in the relevant inventory?

sbearcsiro reviewed (changes requested):
I think this will always install postgis, which needs to be fixed.

I'm not sure what the motivation for all these changes is, seems like any environmental things can be fixed in inventories.

vjrj commented on ansible/roles/postgresql/tasks/postgis.yml:
```diff
@@ -46,9 +46,9 @@
-    - "/usr/share/postgresql/{{pg_version}}/contrib/postgis-{{postgis_version}}/postgis.sql"
-    - "/usr/share/postgresql/{{pg_version}}/contrib/postgis-{{postgis_version}}/spatial_ref_sys.sql"
-    - "/usr/share/postgresql/{{pg_version}}/contrib/postgis-{{postgis_version}}/postgis_comments.sql"
+    - "/usr/share/postgresql/{{pg_version}}/contrib/postgis-{{postgis_scripts_version}}/postgis.sql"
```
Ok. Pending

vjrj commented:
> I think this will always install postgis, which needs to be fixed.

You are right. I added an extra var `use_postgis` just added to `images/spatial/phylolink` so `postgis` is only installed there.

`doi-service` install:

![image](https://user-images.githubusercontent.com/180085/183059765-4dbeea4a-938b-4af8-acb1-56296ccf010b.png)

`spatial` install:

![image](https://user-images.githubusercontent.com/180085/183059793-07bdf6ac-7341-4a81-bd06-fc688c76787a.png)
 
> I'm not sure what the motivation for all these changes is, seems like any environmental things can be fixed in inventories.

The motivation is the same as the `tomcat` or `mysql` roles and facts to install automatically them without too much pain in any ubuntu recent version. You can still force a `tomcat` version or a mysql version (or with this PR the `pg` version), but if not, the `mysql` and `tomcat` default versions are selected from the facts depending of the OS.

A typical problem is when people try to install some service that uses `postgres` and `ansible` fails with a some pg package not found because the default `pg` version does not match the OS version. I added vars like `data_quality_pg_version` in the past to try to avoid that. But trying to give support to newer versions (see #598 #599) I realized that is better to do this via common/set-facts.

sbearcsiro commented:
> A typical problem is when people try to install some service that uses `postgres` and `ansible` fails with a some `pg` package not found because the default `pg` version does not match the OS version. I added vars like `data_quality_pg_version` in the past to try to avoid that. But trying to give support to newer `ubuntu` versions (see #598 #599) I realized that is better to do this via `common/set-facts.`

ok, that makes sense.  it would be nice if a playbook could specify a minimum pg version I guess but other than that, this looks fine.
